components/external_storage/src/lib.rs

```diff
@@ -306,6 +308,7 @@ where
             })?;
         }
         file_length += bytes_read as u64;
+        yield_checker.check().await;
```

Comment:
I'm not found the relative resource-limiter code, which service use this function?

Reply:
Used by `SstService::download`, br uses this function to do download ssts files. Because we already wrap the outer future task with `LimitedFuture`, we just need to let the inner future yield if it already eats too much cpu time.

Reply:
Get it. so the `SStService::download` used by `backup` and  `read_external_storage_into_file`?

Reply:
yes. backup uses `SStService::download` to download external sst files, and it calls the `read_external_storage_into_file` that actually do the job.